pkg/cosign/tuf/client.go

```diff
@@ -432,9 +437,38 @@ func (t *TUF) updateMetadataAndDownloadTargets() error {
 		}
 		b, innerErr := json.MarshalIndent(status, "", "\t")
 		if innerErr != nil {
-			return innerErr
+			return nil, innerErr
 		}
-		return fmt.Errorf("error updating to TUF remote mirror: %w\nremote status:%s", err, string(b))
+		return nil, fmt.Errorf("error updating to TUF remote mirror: %w\nremote status:%s", err, string(b))
+	}
+	// Success! Cache new metadata, if needed.
+	if noCache() {
+		return targets, nil
+	}
+	// Sync the in-memory local store to the on-disk cache.
+	tufDB := filepath.FromSlash(filepath.Join(rootCacheDir(), "tuf.db"))
+	diskLocal, err := tuf_leveldbstore.FileLocalStore(tufDB)
```

Comment:
Is this API expected to be used, or are we reaching into private APIs? I just wanna make sure this is maintainable.